src/domain/place/placeInfo/PlaceMapLink.tsx

```diff
@@ -0,0 +1,59 @@
+import { Button, IconMap } from 'hds-react';
+import React from 'react';
+
+import styles from './placeInfo.module.scss';
+
+export interface PlaceMapLinkEntryProps {
+  id: string;
+  label: string;
+  description?: string;
+  url: string;
+}
+
+export interface PlaceMapLinkProps extends PlaceMapLinkEntryProps {
+  variant?: 'a' | 'button';
+}
+
+const PlaceMapLink: React.FC<PlaceMapLinkProps> = ({
+  id,
+  label,
+  description,
+  url,
+  variant = 'button',
+}) => {
+  // <a> -element variant
+  if (variant === 'a') {
+    return (
+      <>
+        <a
+          id={id}
+          href={url}
+          rel="noopener noreferrer"
+          target="_blank"
+          className={styles.linkEntry}
+        >
+          <IconMap className={styles.linkIcon} />
+          {label}
+        </a>
+        {description && <p className={styles.linkDescription}>{description}</p>}
+      </>
+    );
+  }
+
+  // <button> -element variant
+  return (
+    <>
+      <Button
+        id={id}
+        onClick={() => window.open(url)}
+        iconLeft={<IconMap />}
+        variant="supplementary"
+      >
+        {label}
+      </Button>
+      {description && <p className={styles.linkDescription}>{description}</p>}
+    </>
+  );
+};
```

Comment:
I was playing little bit with this code to find out if there is a way to make it little bit more concise and came up with this:

```
const PlaceMapLink: React.FC<PlaceMapLinkProps> = ({
  id,
  label,
  description,
  url,
  variant = 'button',
}) => {
  const Component = { a: 'a', button: Button }[variant];
  const isLink = variant === 'a';

  const props = {
    a: {
      href: url,
      rel: 'noopener noreferrer',
      target: '_blank',
      className: styles.linkEntry,
    },
    button: {
      onClick: () => window.open(url),
      iconLeft: <IconMap />,
      variant: 'supplementary' as const,
    },
  }[variant];

  return (
    <>
      <Component id={id} {...props}>
        {isLink && <IconMap className={styles.linkIcon} />}
        {label}
      </Component>
      {description && <p className={styles.linkDescription}>{description}</p>}
    </>
  );
};
```

Reply:
Seems to work and event typescript is happy. What do you think? At least description and <> is not repeated twice there :D

Reply:
> I was playing little bit with this code to find out if there is a way to make it little bit more concise and came up with this:

At least that is a very interesting view 😃 I find it harder to read, although the rendering is more compact. We could ask 3rd opinion!

Reply:
at least <Button /> and <a /> parts could be renderered(returned) in/from separate function?

Reply:
the section that actually differs from each other

Reply:
I was thinking that I could make a component factory there also :D ..or at least split those 2 solutions to 2 different components.

Reply:
@Sartonon I took your idea and tried to edit it so that there would be less references to variant. Now there is only 1 path selection and 1 render. I'm happy with it like this! 🥳

Reply:
Great it is looking good!